async-nats/src/jetstream/message.rs

```diff
@@ -561,3 +537,53 @@ pub struct Info<'a> {
     /// Optional token, present in servers post-ADR-15
     pub token: Option<&'a str>,
 }
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum AckErrorKind {
+    RequestFailed,
+    TimedOut,
+    PrefixNotFound,
+    SubjectNotFound,
+    NoReplySubject,
+    SubscriptionDropped,
+}
+
+impl Display for AckErrorKind {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::RequestFailed => write!(f, "failed to publish acknowledge"),
+            Self::NoReplySubject => write!(f, "no reply subject, not a JetStream message"),
+            Self::SubjectNotFound => write!(f, "did not found reply subject"),
+            Self::PrefixNotFound => write!(f, "did not found proper prefix"),
+            Self::TimedOut => write!(f, "double ack response timed out"),
+            Self::SubscriptionDropped => write!(f, "subscription dropped"),
+        }
+    }
+}
+
+pub type AckError = crate::error::Error<AckErrorKind>;
+
+#[derive(Clone, Copy, Debug, PartialEq)]
```

Comment:
All those errors mean something really bad happened internally in NATS protocol.
There is no point for the user to consider those variants, and there is nothing they can do to fix it or prevent it.